includes/template-library/sources/remote.php

```diff
@@ -210,6 +216,91 @@ public function get_data( array $args, $context = 'display' ) {
 		return $data;
 	}
 
+	/**
+	 * Get templates data from transient or by remote request.
+	 * The remote request will happen in one of those 3 conditions:
+	 * 1. Force update.
+	 * 2. The data saved in the transient is empty.
+	 * 3. The container feature has changed since the last data fetch.
+	 *
+	 * @param $force_update
+	 * @return array
+	 */
+	private function get_templates_data( $force_update ) {
+		$templates_data_cache_key = static::TEMPLATES_DATA_TRANSIENT_KEY_PREFIX . ELEMENTOR_VERSION;
+		$templates_data_layout_type_cache_key = static::TEMPLATES_DATA_LAYOUT_TYPE_TRANSIENT_KEY;
+
+		$experiments_manager = Plugin::$instance->experiments;
+		$editor_layout_type = $experiments_manager->is_feature_active( 'container' ) ? 'container_flexbox' : '';
+
+		if ( $force_update ) {
+			return $this->get_templates( $editor_layout_type );
+		}
+
+		$cached_templates_editor_layout_type = get_transient( $templates_data_layout_type_cache_key );
+
+		if ( $cached_templates_editor_layout_type !== $editor_layout_type ) {
+			return $this->get_templates( $editor_layout_type );
+		}
+
+		$templates_data = get_transient( $templates_data_cache_key );
```

Comment:
As I mentioned earlier, performance-wise I'm not sure I like that the DB is called twice in the same function.
Together with that, keeping all the template data in the memory if not needed is also unnecessary, so I'm not sure what's better here.

Reply:
Same as above.